### DISCUSSION TITLE
entry function is decompiled improperly

### DISCUSSION THREAD

**Discussion by elihawila** [NationalSecurityAgency/ghidra#5809] [Q&A]
I'm currently taking a malware analysis course and as part of the course I'm analyzing the malware file re_lotsofstuff.exe with Ghidra. Ghidra on my machine is not producing the decompiled code of the entry function that is shown in the lecture. Please let me know what can I do to fix it.

**Screenshots**

How it looks like in the lecture: 
![screenshot 1](https://github.com/NationalSecurityAgency/ghidra/assets/139172260/91b9efae-6bc5-417e-87cd-c3e10362e033)

How it looks like on my machine:
![screenshot2](https://github.com/NationalSecurityAgency/ghidra/assets/139172260/9e572be0-d240-40a9-ac6b-d052aec4d9ca)

**Environment (please complete the following information):**
 - OS: Windows 10 (Flare-VM)
 - Ghidra Version: 10.3.3
 - Ghidra Origin: Official GitHub distro

**Comment by ryanmkurtz** [comment]
Can you attach your binary?

**Comment by astrelsky** [comment]
The difference here is that Ghidra has improved and was able to identify `__tmainCRTStartup` and treated the jmp as a call return. If you want to see similar output to the lecture slide, right click on `JMP __tmainCRTStartup`  select the flow override option and choose branch.

Note that decompiled output would almost never be the same between different versions of Ghidra. This is due to decompiler improvements (hopefully not regressions)

**Comment by jobermayr** [comment]
Maybe in first Screenshot also a patched version was used. Plain Ghidra doesn't show InterlockedCompareExchange in Decompiler view (#5788).
